packages/nimble-components/src/select/index.ts

```diff
@@ -67,27 +201,164 @@ export class Select extends FoundationSelect implements ErrorPattern {
     ): void {
         const value = this.value;
         super.slottedOptionsChanged(prev, next);
+        this.filterOptions();
         if (value) {
             this.value = value;
         }
     }
 
-    private regionChanged(
-        _prev: AnchoredRegion | undefined,
-        _next: AnchoredRegion | undefined
-    ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+    public override clickHandler(e: MouseEvent): boolean {
+        this.updateSelectedIndexFromFilteredSet();
+        super.clickHandler(e);
+        return true;
+    }
+
+    public inputClickHandler(e: MouseEvent): void {
+        e.stopPropagation(); // clicking in filter input shouldn't close dropdown
+    }
+
+    /**
+     * Handle content changes on the control input.
+     *
+     * @param e - the input event
+     * @internal
+     */
+    public inputHandler(e: InputEvent): boolean {
+        this.filter = this.searchInput?.value ?? '';
+        if (!this.committedSelectedOption) {
+            this.committedSelectedOption = this._options.find(
+                option => option.selected
+            );
+        }
+        this.filterOptions();
+        this.clearSelection();
+        if (this.committedSelectedOption) {
+            this.committedSelectedOption.selected = true;
+        }
+
+        if (
+            this.filteredOptions.length > 0
+            && this.committedSelectedOption
+            && !this.filteredOptions.includes(this.committedSelectedOption)
+        ) {
+            this.committedSelectedOption.selected = false;
+            this.filteredOptions[0]!.selected = true;
+        }
+
+        if (e.inputType.includes('deleteContent') || !this.filter.length) {
+            return true;
         }
+
+        e.stopPropagation();
+        return true;
     }
 
-    private controlChanged(
-        _prev: HTMLElement | undefined,
-        _next: HTMLElement | undefined
+    // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
```

Comment:
Can you add a short explanation of why you needed to disable this line rule? Something about how we've set the return type here to match the return type of the super class's keydownHandler.